cpu/stm32/include/cpu_conf.h

```diff
@@ -116,21 +116,69 @@ extern "C" {
 #define FLASHPAGE_SIZE                  (128U)
 #endif
 
+#ifdef FLASHPAGE_SIZE
 #define FLASHPAGE_NUMOF                 (STM32_FLASHSIZE / FLASHPAGE_SIZE)
+#endif
+
+#if defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || \
+    defined(CPU_FAM_STM32F7)
+#define PERIPH_FLASHPAGE_CUSTOM_PAGESIZES
+
+/**
+ * @brief stm32 dual bank configuration
+ *
+ * By default, the stm32f4 series with 1MB flash enable the DB1M flag to split
+ * the 1MB flash into two banks, 2MB devices are always split in two banks.
+ * On both the stm32f4 and the stm32f7 this can be modified with user
+ * programmable flags. Detecting the settings at runtime is not supported
+ *
+ * @note This must match the setting on the MCU. by default it is assumed that
+ * the user has not changed this setting manually.
+ */
+#if (defined(FLASH_OPTCR_DB1M) && (STM32_FLASHSIZE >= (1024 * 1024)))
+#define FLASHPAGE_DUAL_BANK             1
+#else
+#define FLASHPAGE_DUAL_BANK             0
+#endif
+
+/* stm32f7 uses single bank with 32KB to 256KB sectors on 2MB devices */
+#if defined(CPU_FAM_STM32F7) && (STM32_FLASHSIZE == (2048 * 2024))
+#define FLASHPAGE_MIN_SECTOR_SIZE       (32 * 1024)
+#else
+#define FLASHPAGE_MIN_SECTOR_SIZE       (16 * 1024)
+#endif
+
+#if FLASHPAGE_DUAL_BANK
+/* Number of "large" sectors + 4 for the small sectors that together equal a
+ * single large sector. Times two to account for the two banks */
+#define FLASHPAGE_NUMOF                 ((STM32_FLASHSIZE / \
+                                         (8 * FLASHPAGE_MIN_SECTOR_SIZE)) + 8)
```

Comment:
```suggestion
                                         (8 * 2* FLASHPAGE_MIN_SECTOR_SIZE)) + 8)
```

To account for the dual bank

Reply:
I think I'm completely off on this comment. I misread the define.